File: src/main/java/com/saybetter/global/auth/info/GoogleOAuth2UserInfo.java
package com.saybetter.global.auth.info;

import java.util.Map;

import lombok.AllArgsConstructor;

@AllArgsConstructor
public class GoogleOAuth2UserInfo implements OAuth2UserInfo {

	private Map<String, Object> attributes;

	@Override
	public String getProviderId() {
		return (String)attributes.get("sub");
	}

	@Override
	public String getProvider() {
		return "google";
	}

	@Override
	public String getEmail() {
		return (String)attributes.get("email");
	}

	@Override
	public String getName() {
		return (String)attributes.get("name");
	}
}

File: core/core-api/src/main/java/io/say/better/global/auth/info/GoogleOAuth2UserInfo.kt
package io.say.better.global.auth.info

data class GoogleOAuth2UserInfo(
    val attributes: Map<String, Any>?
) : OAuth2UserInfo {

    override val providerId: String
        get() = (attributes!!["sub"] as String)

    override val provider: String
        get() = "google"

    override val email: String
        get() = attributes!!["email"] as String

    override val name: String
        get() = attributes!!["name"] as String
}
